src/Nethermind/Nethermind.TxPool/Filters/IIncomingTxFilter.cs

```diff
@@ -0,0 +1,30 @@
+﻿//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using Nethermind.Core;
+
+namespace Nethermind.TxPool.Filters
+{
+    /// <summary>
+    /// Filter used for discarding inbound transactions in the TX pool.
+    /// Name used to differentiate from filters from Consensus namespace.
+    /// </summary>
+    public interface IIncomingTxFilter
+    {
+        (bool Accepted, AddTxResult? Reason) Accept(Transaction tx, TxHandlingOptions txHandlingOptions);
```

Comment:
If going this way we probably don't need Accepted as separate value, we can just use AddTxResult and return Added (we can rename it to Accepted)

Reply:
keeping it consistent with another filter for no until both filters are reviewed

Reply:
I myself don't see the point of keeping this consistency, the only thing we are duplicating returned data.